libs/vkd3d/command.c

```diff
@@ -1998,6 +2000,16 @@ static void d3d12_command_list_begin_new_sequence(struct d3d12_command_list *lis
     d3d12_command_list_invalidate_all_state(list);
     /* Extra special consideration since we're starting a fresh command buffer. */
     list->descriptor_heap.buffers.heap_dirty = true;
+    d3d12_command_list_debug_mark_label(list, "Split", 0.0f, 0.0f, 0.0f, 1.0f);
+}
+
+static void d3d12_command_list_consider_new_sequence(struct d3d12_command_list *list)
+{
+    if (list->cmd.vk_command_buffer == list->cmd.vk_init_commands_post_indirect_barrier &&
+            vkd3d_atomic_uint32_load_explicit(&list->device->device_has_dgc_templates, vkd3d_memory_order_relaxed))
```

Comment:
For correctness, we also need to make sure that there are no scoped queries active (think it's only `PIPELINE_STATISTICS` at this point, which isn't super common anyway). We currently don't have code for that, but just adding a counter to the command list for active query tracking should do.

Reply:
That's what https://github.com/HansKristian-Work/vkd3d-proton/pull/1694/commits/ded0fc4df308d2fbc67e46ac67c2d87160c18d2c does.

Reply:
missed that somehow, derp.